### DISCUSSION TITLE
NGINX (Loadbalancer): Custom configuration

### DISCUSSION THREAD

**Discussion by privateVoit** [caos/orbos#713] [Ideas]
Hi there!

We are currenlty seeing a huge load on one of our clusters (~1700 active connections). The nginx is hitting the worker_rlimit and therefore dropping connections.
To fix that we had to scale the orbiter deployment to zero so we could edit the nginx.conf without node-agent interfering.

We added following directive:
`
worker_rlimit_nofile 8192; 
`
Since this isn't the first time that we needed a change in the nginx.conf a custom configuration for the nginx.conf would be nice.

**Possible solution:**
It would be cool if the node-agent would add follwing line to the nginx configuration. 
`include /etc/nginx/conf.d/*;`

This way the administrator has the ability to add custom configuration in the conf.d directory like `worker_rlimit_nofile`.

Is this something you might consider adding?

Thank you in advance!

**Comment by fforootd** [comment]
Hi @privateVoit 

Thank you for providing this input.

Would it not make more sense to create an option to provide a custom `nginx` config within the clusters repository?
With this the config would benefit from the same `GitOps` benefits!

**Reply by privateVoit** [reply]
@fforootd That's even better! This way we don't have deploy those nginx configurations to all nodes manually.

**Reply by fforootd** [reply]
Ok, we are not yet sure **when** we will implement this feature, but we definitely will consider this in the future.

If you are in a hurry feel free to contact us [hi@caos.ch](mailto:hi@caos.ch)

**Reply by fforootd** [reply]
@eliobischof @stebenz @thesephirot any additional inputs?

I think we should tweak the default config as well.

**Reply by eliobischof** [reply]
our latest node agent always adds the directive `worker_rlimit_nofile 8192;`
